scripts/deploy.sh

```diff
@@ -37,24 +45,32 @@ which openstack-ansible || ./scripts/bootstrap-ansible.sh
 ./scripts/pw-token-gen.py --file /etc/openstack_deploy/user_extras_secrets.yml
 
 # begin the openstack installation
-cd "${OSAD_DIR}"/playbooks/
+if [[ "${DEPLOY_OSAD}" == "yes" ]]; then
+  cd ${OSAD_DIR}/playbooks/
 
-# setup the hosts and build the basic containers
-install_bits setup-hosts.yml
+  # ensure that the ELK containers aren't created if they're not
+  # going to be used
+  if [[ "${DEPLOY_ELK}" != "yes" ]]; then
+    rm -f /etc/openstack_deploy/env.d/{elasticsearch,logstash,kibana}.yml
+  fi
 
-# setup the haproxy load balancer
-if [[ "${DEPLOY_HAPROXY}" == "yes" ]]; then
-  install_bits haproxy-install.yml
-fi
+  # setup the hosts and build the basic containers
+  install_bits setup-hosts.yml
+
+  # setup the haproxy load balancer
+  if [[ "${DEPLOY_HAPROXY}" == "yes" ]]; then
+    install_bits haproxy-install.yml
```

Comment:
I'm against making changes for no reason, on principle - so if there's no reason I'd probably be against it. Although I can see that it would work.